sdks/python/apache_beam/transforms/trigger.py

```diff
@@ -165,7 +166,7 @@ def on_merge(self, to_be_merged, merge_result, context):
     pass
 
   @abstractmethod
-  def should_fire(self, watermark, window, context):
+  def should_fire(self, watermark, window, context, time_domain):
```

Comment:
<!--new_thread; commit:a68ceca8046b6d42f941aa4016a528ec2ee3808a; resolved:0-->
Could you take `time_domain` as the first argument?  Rename `watermark` to `timestamp`, and update the pydoc below to indicate that the timestamp is in the particular time domain.

Reply:
done